doop-analyzer/README.md

```diff
@@ -113,18 +113,148 @@ will be parsed into this:
 }
 ```
 
-### Processing rules
+As with the cluster identity, doop-analyzer does not care about the exact keys and values in the object identity (except
+that object identity must be equal when merging violations, see below), but to make further processing easier, it's
+probably a good idea to keep the set of keys consistent across all violations.
 
-TODO
+### Rule-based rewriting
 
-### Merging rules
+Custom rules can be provided in the configuration in order to process violations based on regex matches. Rewriting
+happens in two phases.
 
-TODO
+#### Processing rules
+
+In the first phase, the **processing rules** from the configuration section `processing_rules` are applied to each
+matching violation. Processing rules rewrite the original violation itself. As a basic example, consider rules that
+apply to Helm releases: As of Helm 3, Helm releases are stored in Secret objects with the name schema
+`sh.helm.release.v1.${NAME}.v${VERSION}`. This implementation detail can be hidden from the user-visible violations by
+rewriting with the following processing rule:
+
+```yaml
+processing_rules:
+  - match: { kind: Secret }
+    replace:
+      source:  name
+      pattern: 'sh\.helm\.release\.v1\.(.*\.v\d+)'
+      target:  { kind: Helm 3 release, name: '$1' }
+```
+
+Applying this processing rule to a violation like this:
+
+```json
+{
+  "kind": "Secret",
+  "name": "sh.helm.release.v1.foo-bar.v1",
+  "namespace": "foo",
+  "message": "I don't like this",
+  "object_identity": {}
+}
+```
+
+will yield the following result:
+
+```json
+{
+  "kind": "Helm 3 release",
+  "name": "foo-bar.v1",
+  "namespace": "foo",
+  "message": "I don't like this",
+  "object_identity": {}
+}
+```
+
+#### Rule syntax
+
+Each processing rule may contain the following fields:
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| `description` | string | A human-readable description of what this rule is about. This field is not interpreted by doop-analyzer at all, but it can be used for documentation purposes. Since it's a structured field instead of just a YAML comment, it is more likely to be preserved when editing rules in a specialized UI. |
+| `match` | object of regexes | If given, the rule will only be applied if, for each key-value pair in this object, the violation has an attribute whose name matches the key, and whose value matches the regex. (In the example above, the rule only applies to violations whose `kind` attribute matches the regex `Secret`.) See below for notes on attribute names and regex syntax. |
```

Comment:
So the regex will only be applied if it finds a match in each key-value pair? Or only to those who match?

Reply:
The quantifier is "for each key-value pair", so the former. I agree that the sentence is rather long, but I don't see any immediate way to make it more compact without losing strictness. I removed a comma to maybe make the statement structure clearer.